lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -30,11 +30,19 @@ function Get-Vcredist {
         try {
             Write-JujuWarning "Trying to get vcredist Juju resource"
             $vcredistPath = Get-JujuResource -Resource "vcredist-x64"
-            return $vcredistPath
         } catch {
+            $vcredistPath = $null
             Write-JujuWarning "Failed downloading vcredist resource: $_"
-            Write-JujuWarning "Falling back to file download"
         }
+        if ($vcredistPath) {
+            $bytes = Get-Content $vcredistPath -TotalCount 31 -Encoding Byte
+            $str = [string]::join("", [char[]]$bytes)
+            if ($str -eq $DEFAULT_JUJU_RESOURCE_CONTENT) {
+                Throw "Manually attach the Juju resource 'vcredist-x64' or set the config option 'vcredist-url'."
```

Comment:
I don't want to force anyone to use resources. I want a more transparent status message of what is going on, and what steps the operator must take to remedy the situation. Set-JujuStatus alongside Write-JujuWarning is not unreasonable.

Also, if you want to make your functions more backwards compatible, consider doing some sanity checks in BEGIN{} section of the function.

```powershell
function Get-Vcredist {
    BEGIN {
        $resourcesSupport = (Get-Command resource-get.exe -ErrorAction SilentlyContinue) -ne $null
    }
    PROCESS {
        Write-JujuWarning "Getting vcredist."
        $location = ""
        if ($resourcesSupport) {
            $location = Get-JujuResource -Resource "vcredist-x64"
            $bytes = Get-Content $vcredistPath -TotalCount 31 -Encoding Byte
            $str = [string]::join("", [char[]]$bytes)
            if ($str -ne $DEFAULT_JUJU_RESOURCE_CONTENT) {
                return $location
            }
        }
        $vcredistUrl = Get-JujuCharmConfig -Scope "vcredist-url"
        if(!$vcredistUrl) {
            $msg = "Coult not find suitable download location for vcredist."
            if($resourcesSupport) {
                $msg += " Consider uploading the resource to your controller"
            } else {
                $msg += " Please set vcredist-url in your config"
            }
            Set-JujuStatus blocked -Message $msg
            return $false
        }
        $file = ([System.Uri]$url).Segments[-1]
        $location = Join-Path $env:TEMP $file
        Start-ExecuteWithRetry {
            Invoke-FastWebRequest -Uri $url -OutFile $tempDownloadFile
        } -RetryMessage "Downloading vcredist failed. Retrying..."

        return $location
}
```